ui/src/app/views/homepage/component.tsx

```diff
@@ -0,0 +1,394 @@
+        </div>
+        <div style={{flexDirection: 'column', width: '50%', padding: '1rem'}}>
+          <div style={{...styles.infoBox, flexDirection: 'column'}}>
+            <div style={{display: 'flex', flexDirection: 'row', justifyContent: 'space-between'}}>
+              <div style={{flexDirection: 'column', width: '70%'}}>
+                <div style={styles.infoBoxHeader}>Login to ERA Commons</div>
+                <div style={styles.infoBoxBody}>Clicking the login link will bring you
+                  to the ERA Commons Portal and redirect you back to the
+                  Workbench once you are logged in.</div>
+              </div>
+              <AccountLinkingButton failed={false}
+                                    completed={this.state.eraCommonsLinked}
+                                    defaultText='Login'
+                                    completedText='Linked'
+                                    failedText='Error Linking Accounts'
+                                    onClick={AccountLinking.redirectToNiH}/>
+            </div>
+            {this.state.eraCommonsError && <Error>
+              <ClrIcon shape='exclamation-triangle' class='is-solid'/>
+              Error Linking NIH Username: {this.state.eraCommonsError} Please try again!
+            </Error>}
+          </div>
+          <div style={{...styles.infoBox, marginTop: '0.7rem'}}>
+            <div style={{flexDirection: 'column', width: '70%'}}>
+              <div style={styles.infoBoxHeader}>Complete Online Training</div>
+              <div style={styles.infoBoxBody}>Clicking the training link will bring you to
+                the All of Us Compliance Training Portal, which will show you any
+                outstanding training material to be completed.</div>
+            </div>
+            <AccountLinkingButton failed={false}
+                                  completed={this.props.trainingCompleted}
+                                  defaultText='Complete Training'
+                                  completedText='Completed'
+                                  failedText=''
+                                  onClick={() => {}}/>
+          </div>
+        </div>
+      </div>
+    </React.Fragment>;
+  }
+}
+
+@Component({
+  selector: 'app-account-linking',
+  template: '<div #root></div>',
+})
+export class AccountLinkingComponent extends ReactWrapperBase {
+  @Input('eraCommonsLinked') eraCommonsLinked: AccountLinkingProps['eraCommonsLinked'];
+  @Input('trainingCompleted') trainingCompleted: AccountLinkingProps['trainingCompleted'];
+  constructor() {
+    super(AccountLinking,
+        ['eraCommonsLinked', 'trainingCompleted']);
+  }
+}
+
+
+@Component({
+  styleUrls: ['./component.css'],
+  templateUrl: './component.html',
+})
+
+export class HomepageComponent implements OnInit, OnDestroy {
+  private static pageId = 'homepage';
+  @ViewChild('myVideo') myVideo: any;
+  profile: Profile;
+  view: any[] = [180, 180];
+  numberOfTotalTasks = 4;
```

Comment:
Existing issue: There's definitely a bunch of dead code here - do you mind putting a TODO and filing a ticket for this cleanup (including my other comment around the double profile calls)

Reply:
that cleanup can probably be done when converting the homepage component.  I'll make a note in that ticket.

Reply:
sounds good - thanks